### PR TITLE
fix(android): Fix system keyboard globe button override

### LINKED ISSUE #3101: [Android] Fix system keyboard Globe button actions
[KMManager.setGlobeKeyAction()](https://help.keyman.com/DEVELOPER/ENGINE/android/13.0/KMManager/setGlobeKeyAction) can set globe actions to be
```java
  // Globe key actions
  public enum GlobeKeyAction {
    GLOBE_KEY_ACTION_SHOW_MENU,
    GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD,         // Switch to next Keyman keyboard
    GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD, // Advance to next system keyboard
    GLOBE_KEY_ACTION_DO_NOTHING,
  }
```

The change introduced in #1709 (cycling through keyboards on lock screen) reverts the system keyboard globe button to the default (`GlobeKeyAction.GLOBE_KEY_ACTION_SHOW_MENU`).

We need a way to allow keyboard creators to set the globe key action.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #3101
Use a different state variable while the screen is locked so that `sysKbGlobeKeyAction` is preserved.

### User Testing
Note: This test involves locking the screen with a password (Latin characters). Out of precaution,  it is best tested in an emulator or if using a physical device, have a secondary unlock (e.g. fingerprint).
- [ ] Verify system keyboard can switch while screen is locked
1. Load Keyman and enable as a system keyboard
2. Install a non-Latin keyboard (e.g. khmer_angkor)
3. Have the system keyboard set to the non-Latin keyboard
4. Lock the device
5. Attempt to unlock the device
   a. Initial keyboard would be khmer_angkor
   b. Hitting the globe key should switch to next keyboard (sil_euro_latin)
   c. Hitting the globe key should switch to next keyboard (khmer_angkor)
   d. Hitting the globe key should advance to the next system keyboard (e.g. Gboard with English)
6. Use the English keyboard to unlock the device.

Using a debugger, I can halt in KMManager at the block
```
switch(action) {
```
and verify `action` can have a different value from `sysKbGlobeKeyAction`.

### CODE REVIEW

**Comment by darcywong00** [comment]
Note: the CI test build currently fails on master because of a mismatched version in `/common/core/web/utils/package.json`

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java [hunk 1]
```diff
@@ -1816,28 +1816,36 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
         if (KMManager.shouldAllowSetKeyboard()) {
           if (SystemKeyboard.keyboardPickerEnabled) {
             KeyguardManager km = (KeyguardManager) appContext.getSystemService(Context.KEYGUARD_SERVICE);
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:1818]:
```suggestion
            KeyguardManager kgm = (KeyguardManager) appContext.getSystemService(Context.KEYGUARD_SERVICE);
```

`km` is too close to `Keyman` :)

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java [hunk 2]
```diff
@@ -1816,28 +1816,36 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
...
             if(km.inKeyguardRestrictedInputMode()) {
               // Override system keyboard globe key action if screen is locked:
               // 1. Switch to next Keyman keyboard (no menu)
               // 2. When all the Keyman keyboards have been cycled through, advance to the next system keyboard
-              if (sysKbIndexOnLockScreen == -1) {
-                sysKbIndexOnLockScreen = getCurrentKeyboardIndex(context);
-                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD;
-              } else if (sysKbIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
-                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD;
+              if (sysKbIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:1824]: When would this be `true`? Wouldn't `getCurrentKeyboardIndex()` return the current keyboard that's been changed anyway? Or doesn't `switchToNextKeyboard()` alter that?

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:1824]: getCurrentKeyboardIndex() does return the current keyboard (as it's been cycled through `GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD`. 

When it goes all the way around the cycle, (`sysKbIndexOnLockScreen = getCurrentKeyboardIndex()` and the action becomes `GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD`

Comment (mcdurdin) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:1824]: Ah snap. It's the `sysKbIndexOnLockScreen` that doesn't change. Perhaps that should be renamed to `sysKbStartingIndexOnLockScreen` or something like that, to clarify?

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java (hunk 1 above)
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java]: I went with a longer `keyguardManager` for disambiguation :)

**Pull request merged**